### ISSUE TITLE
Statistics Review: bayes_mvs (Trac #147)

### ISSUE THREAD

**Issue by scipy-gitbot** [scipy/scipy#674]
_Original ticket http://projects.scipy.org/scipy/ticket/147 on 2006-04-02 by @rkern, assigned to @rkern._

The function `bayes_mvs` in file source:trunk/Lib/stats/morestats.py needs review.

Please look over the StatisticsReview guidelines and add your comments below.

**Comment by aeklant** [comment]
@rgommers ping

I would add an example to the docstring. Other than that I'm not sure the documentation is lacking.

There are virtually no test cases so I will add at least a base one from the example and one for empty input (although it is properly handled with a  `ValueError` due to insufficient data).

As far as nan input, it is the same as most other functions, returning nan by default so I plan to add the `nan_check` keyword.

**Comment by rgommers** [comment]
A See Also section in the docstring that refers to ``mvsdist`` (and vice versa) would be useful, in addition to the example. Other than that the docstring indeed looks quite good.

Adding a basic test for ``bayes_mvs`` is enough I think; the code is very straightforward. However, the interesting part of the code is in ``mvsdist``, which is also almost complete untested. So adding better test coverage there would be more interesting.

**Comment by aeklant** [comment]
Please review gh-5065

**Comment by rgommers** [comment]
The paper on which ``bayes_mvs`` and ``mvsdist`` are based is at least peer-reviewed, but only has 2 citations. I suspect that if these functions were submitted for inclusion now we wouldn't accept them. However, they do look useful and the implementation seems to be OK. 

I've checked that around ``n=1000`` the approximation is OK, i.e. no big jumps when plotting mean/std/var against ``n_samples = np.arange(980, 1020)``. Relative accuracy of std/var is around 1e-6 here:

    n_repeats = 1  # Useful when using randomly generated data below
    res = []
    for n_samples in numvals:
        means = []
        for ii in range(n_repeats):
            data = np.linspace(0.344, 0.346, num=n_samples)
            res_mean, res_var, res_std = stats.bayes_mvs(data, alpha=0.95)
            means.append(res_var.statistic)

        res.append(np.mean(means))

    print('Relative difference around n=1000: %s' % (np.diff(res).max() / res[0]))


And that for random samples from a normal distribution, it does get its estimates of mean/std/var right. I have checked only the implementation of ``mdist`` against the paper, took a bit of puzzling but it's correct. I then trust that the other ones will also be correct.
